go/ms/internal/sqlite/db.go

```diff
@@ -0,0 +1,226 @@
+	e.RLock()
+	defer e.RUnlock()
+	rows, err := e.db.QueryContext(ctx, FullMapEntryByIP, ip)
+	if err != nil {
+		return nil, serrors.Wrap(db.ErrReadFailed, err)
+	}
+	defer rows.Close()
+	got := []FullMapRow{}
+	for rows.Next() {
+		var r FullMapRow
+		err = rows.Scan(&r.Id, &r.IP, &r.IA, &r.Timestamp)
+		if err != nil {
+			return nil, serrors.Wrap(db.ErrDataInvalid, err)
+		}
+		got = append(got, r)
+	}
+	return got, nil
+}
+
+//InsertFullMapEntry inserts a row into the full_map table
+func (e *executor) InsertFullMapEntry(ctx context.Context, fmRow FullMapRow) (sql.Result, error) {
+	//TODO (supraja): handle transaction correctly here
+	res, err := e.db.ExecContext(ctx, InsFullMapEntry, fmRow.IP, fmRow.IA, fmRow.Timestamp)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+//DeleteFullMapEntryById deletes a row in the full_map table by id
+func (e *executor) DeleteFullMapEntryById(ctx context.Context, id int) (sql.Result, error) {
+
+	//TODO (supraja): handle transaction correctly here
+	res, err := e.db.ExecContext(ctx, DelFullMapEntry, id)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+//InsertNewEntry inserts a row into the new_entries table
+func (e *executor) InsertNewEntry(ctx context.Context, entry []byte) (sql.Result, error) {
+
+	//TODO (supraja): handle transaction correctly here
+	res, err := e.db.ExecContext(ctx, InsNewEntry, entry)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+//InsertPCNRep inserts a row into the pcn_reps table
+func (e *executor) InsertPCNRep(ctx context.Context, entry []byte) (sql.Result, error) {
+
+	//TODO (supraja): handle transaction correctly here
+	res, err := e.db.ExecContext(ctx, InsPCNRep, entry)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+//GetNewEntryById queries the new_entries table by id to return the signedPld stored in it
+func (e *executor) GetNewEntryById(ctx context.Context, id int) (*ctrl.SignedPld, error) {
+	e.RLock()
+	defer e.RUnlock()
+	rows, err := e.db.QueryContext(ctx, NewEntryById, id)
+	if err != nil {
+		return nil, serrors.Wrap(db.ErrReadFailed, err)
+	}
+	defer rows.Close()
+
+	cols, err := rows.Columns()
+	if err != nil {
+		fmt.Println("Failed to get columns", err)
+		return nil, err
+	}
+
+	rawResult := make([][]byte, len(cols))
+	got := &ctrl.SignedPld{}
+	dest := make([]interface{}, len(cols)) // A temporary interface{} slice
+
+	for i := range rawResult {
+		dest[i] = &rawResult[i] // Put pointers to each string in the interface slice
+	}
+
+	for rows.Next() {
+		err = rows.Scan(dest...)
+		if err != nil {
+			fmt.Println("Failed to scan row", err)
+			return nil, err
+		}
+	}
+
+	//can use index 0, we expect only one entry to be returned as id is the primary key
+	proto.ParseFromRaw(got, rawResult[0])
+	return got, nil
+}
+
+//GetNewEntries returns all rows of the new_entries table
+func (e *executor) GetNewEntries(ctx context.Context) ([]*ctrl.SignedPld, error) {
```

Comment:
>GetNewEntries

Accessing the columns and rows is duplicated with `GetNewEntryById`.

Why are you using a slice of empty interfaces? To avoid the type conversion from Scan?

Reply:
I extracted the common code into a method now. And yes because we only read [][]byte from it

Reply:
Thanks.

Using `interface{}` in `getRawResultFromRows` should not be necessary and you could even avoid the copy, from the docs:
```
If a dest argument has type *[]byte, Scan saves in that argument a copy of the corresponding data. The copy is owned by the caller and can be modified and held indefinitely. The copy can be avoided by using an argument of type *RawBytes instead; see the documentation for RawBytes for restrictions on its use. 
```

https://golang.org/pkg/database/sql/#Rows.Scan

Reply:
I remember trying to use RawBytes and I had some errors. I will change it for now before I merge but add a TODO comment on that line so I can test it concretely with the rest of the code. Or would you suggest I add unit tests there?